lib/rubygem_fs.rb

```diff
@@ -0,0 +1,78 @@
+module RubygemFs
+
+  def self.instance
+    @fs ||=
+      if Rails.env.development?
+        RubygemFs::Local.new
+      else
+        RubygemFs::S3.new
+      end
+  end
+
+  def self.mock!
+    @fs = RubygemFs::Local.new.tap do |fs|
+      def fs.base_dir
+        @dir ||= Dir.mktmpdir
+      end
+    end
+  end
+
+  def self.s3!(host)
+    uri = URI(host)
+    @fs = RubygemFs::S3.new
+    s3 = @fs.s3(access_key_id: 'k', secret_access_key: 's',  proxy: {host: uri.host, port: uri.port})
```

Comment:
this `s3!` method is only used when you pass a ENV var `S3_PROXY` . Thats so we can simulate using fake-s3 in local development. The key and secret are just mocks on fake-s3.